charts/newrelic-infrastructure/templates/rolebinding.yaml

```diff
@@ -0,0 +1,35 @@
+{{- if .Values.rbac.create }}
+{{- if eq .Values.rbac.limitSecretAccess true }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  labels: {{ include "newrelic.labels" . | indent 4 }}
+  name: {{ template "newrelic.fullname" . }}-secrets
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "newrelic.fullname" . }}-secrets
+subjects:
+- kind: ServiceAccount
+  name: {{ template "newrelic.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace }}
+{{- if .Values.etcdTlsSecretNamespace}}
```

Comment:
I don't get why these are two different cases. How are we going to scope the secret (`limitSecretAccess == true`) if `etcdTlsSecretNamespace` is not defined?

Couldn't we just replace the whole `limitSecretAccess` flag by a `etcdTlsSecretNamespace != ""` check?

Reply:
You are right, the change is safe enough to be done without introducing a new flag.

By the way we should change as well the static manifests in the nri-kubernetes repo